app/src/main/java/com/poupa/vinylmusicplayer/ui/activities/SettingsActivity.java

```diff
@@ -283,9 +301,32 @@ void invalidateSettings() {
                 });
             }
 
-            final ATEColorPreference primaryColorPref = findPreference("primary_color");
+            final TwoStatePreference rememberLastTab = findPreference(PreferenceUtil.REMEMBER_LAST_TAB);
+            if (rememberLastTab != null) {
+                rememberLastTab.setChecked(PreferenceUtil.getInstance().rememberLastTab());
+                rememberLastTab.setOnPreferenceChangeListener((preference, newValue) -> {
+                    // Save preference
+                    PreferenceUtil.getInstance().setRememberLastTab((Boolean) newValue);
+
+                    return true;
+                });
+            }
+
+            final TwoStatePreference whitelistEnabled = findPreference(PreferenceUtil.WHITELIST_ENABLED);
+            if (whitelistEnabled != null) {
+                whitelistEnabled.setChecked(PreferenceUtil.getInstance().getWhitelistEnabled());
+                whitelistEnabled.setOnPreferenceChangeListener((preference, newValue) -> {
+                    // Save preference
+                    //PreferenceUtil.getInstance().setRememberShuffle((Boolean) newValue);
+
+                    return true;
+                });
+            }
+
+            final ATEColorPreference primaryColorPref = findPreference(ThemeStore.KEY_PRIMARY_COLOR);
             if (getActivity() != null && primaryColorPref != null) {
-                final int primaryColor = ThemeStore.primaryColor(getActivity());
+                //final int primaryColor = ThemeStore.primaryColor(getActivity());
+                final int primaryColor = PreferenceUtil.getInstance().getPrimaryColor();
```

Comment:
As you can see here I also had some trouble with the ThemeStore, because the ThemeStore provides a seperate shared pref file, so I wanted to sync the ThemeStore with the main shared prefs, but I don't know wether this is a good approach.